addons/account/models/account_move.py

```diff
@@ -1439,14 +1438,16 @@ def compute_amount_fields(self, amount, src_currency, company_currency, invoice_
         """ Helper function to compute value for fields debit/credit/amount_currency based on an amount and the currencies given in parameter"""
         amount_currency = False
         currency_id = False
+        date = self.env.context.get('date', False)
```

Comment:
Same thing for all other occurrences of this pattern: default `None` should do, why pass a default `False`? ;)